src/externalized/Soldier.cc

```diff
@@ -273,118 +274,81 @@ static void showGearRemoveMessage(const SOLDIERTYPE* s, uint16_t usItem)
 	ScreenMsg(FONT_MCOLOR_LTYELLOW, MSG_INTERFACE, st_format_printf(*GCM->getNewString(NS_SOLDIER_REMOVES_ITEM), s->name, ItemNames[usItem]));
 }
 
+const std::vector<ITEMDEFINE> HEAD_GEARS_DAY   { SUNGOGGLES };
+const std::vector<ITEMDEFINE> HEAD_GEARS_NIGHT { UVGOGGLES, NIGHTGOGGLES };
+
 void Soldier::putNightHeadGear()
 {
-	int8_t nightGogglesPos = FindObj(mSoldier, NIGHTGOGGLES);
-	int8_t uvGogglesPos    = FindObj(mSoldier, UVGOGGLES);
-	int8_t sunGogglesPos   = FindObj(mSoldier, SUNGOGGLES);
-	int8_t freeHeadSlot    = getFreeHeadSlot();
-	bool gasMaskEquiped    = IsWearingHeadGear(*mSoldier, GASMASK);
+	switchHeadGear(HEAD_GEARS_DAY, HEAD_GEARS_NIGHT);
+}
 
+void Soldier::putDayHeadGear()
+{
+	switchHeadGear(HEAD_GEARS_NIGHT, HEAD_GEARS_DAY);
+}
 
-	if(isHeadPosition(uvGogglesPos))
+void Soldier::switchHeadGear(std::vector<ITEMDEFINE> fromGears, std::vector<ITEMDEFINE> toGears)
```

Comment:
Should be passed by reference or just made take a bool to toggle which way to switch.